mantle/platform/qemu.go

```diff
@@ -306,6 +316,64 @@ func (inst *QemuInstance) SwitchBootOrder() (err error) {
 	return nil
 }
 
+// RemovePrimaryBlockDevice deletes the primary device from a qemu instance
+// and sets the seconday device as primary. It expects that all block devices
+// are mirrors.
+func (inst *QemuInstance) RemovePrimaryBlockDevice() (err error) {
+	var primaryDevice string
+	var secondaryDevicePath string
+	monitor, err := newQMPMonitor(inst.tempdir)
+	if err != nil {
+		return errors.Wrapf(err, "Could not connect to QMP device")
+	}
+	if err := monitor.Connect(); err != nil {
+		return errors.Wrapf(err, "Could not connect to QMP device")
+	}
+
+	defer func() {
+		e := monitor.Disconnect()
+		if err != nil {
+			err = fmt.Errorf("%v; %v", err, e)
```

Comment:
I think the `err` defined when `:=` assigning the result of `newQMPMonitor` might shadow the `err` in the function signature, causing us to still return `nil` in this case.  Probably best to rename the one in the signature and the assignments here.

Reply:
yeah, makes sense. In that case, [this](https://github.com/coreos/coreos-assembler/pull/1880/files/430948c58a61000e6e3bb04fa81b1f10f6a5fe08#diff-16a92b89e4fdab63ff64b808b2495d53843ca8e5b6614bb5493be3077c8726ccL258) will also change, right?